cmd/tailscale/cli/serve_v2.go

```diff
@@ -538,34 +538,22 @@ func (e *serveEnv) applyTCPServe(sc *ipn.ServeConfig, dnsName string, srcType se
 		return fmt.Errorf("invalid TCP target %q", target)
 	}
 
-	dstURL, err := url.Parse(target)
+	targetURL, err := expandProxyTargetDev(target, []string{"tcp"}, "tcp")
```

Comment:
optional: you can make the function take a variadic string argument, and just default to the first one. So you don't have to write `[]string{"tcp"}, "tcp"` but just instead `"tcp"`

Reply:
Would just need to add a length check and a comment on the function on what those new arguments do

Reply:
I think that would be nice for the single scheme case, I find the caller hard to grok when there are multiple:

`expandProxyTargetDev("8080", []string{"http", "https", "https+insecure"}, "http")`

vs

`expandProxyTargetDev("8080", "http", "http", "https", "https+insecure")`

Reply:
it would just be `expandProxyTargetDev("8080", "http", "https", "https+insecure")`

but all good either way 👍🏼